src/context/theme-provider.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+
+export interface ThemeProviderProps {
+  children: React.ReactNode
+}
+
+export type Theme = 'light' | 'dark'
+
+export type ThemeContextValues = {
+  currentTheme: Theme,
+  toggleTheme: () => void;
+  toggleThemeWithKey: (event: React.KeyboardEvent<HTMLButtonElement>) => void;
```

Comment:
I'm not convinced we need both of these. Can `toggleTheme` accept an optional event and handle expected keys?

Reply:
i'll try refactoring this.